code/datums/gamemode/dynamic/dynamic_rulesets.dm

```diff
@@ -116,13 +116,22 @@
 	return FALSE
 
 /datum/dynamic_ruleset/proc/get_weight()
+	var/weight = src.weight
 	weight *= weight_time_day()
-	if(repeatable && weight > 1)
```

Comment:
One thing more, not sure if this is intended but you've done 2 things here
* You're only halving weight for repeatable modes, so this won't halve weight for things like nuclear assault. Which seems like the kind of mode people fatigue from the fastest.
* You've removed the >1, which while it isn't a huge change isn't documented. This prevents some modes that are already very low weight from dropping below 1

Reply:
I changed all rulesets to have weight = 1 a while back. I mentioned this briefly in the OP. In light of that, I think that check is no longer necessary. Is there anything that can further lower the weight that I'm missing?
You're right about the check for `repeatable`, that is an oversight and I'll fix it.

Reply:
Futureproofing in case we ever go away from flat weight, I suppose